Content.Server/Nutrition/EntitySystems/DrinkSystem.cs

```diff
@@ -322,7 +325,24 @@ private void OnDoAfter(Entity<DrinkComponent> entity, ref ConsumeDoAfterEvent ar
 
         _audio.PlayPvs(entity.Comp.UseSound, args.Target.Value, AudioParams.Default.WithVolume(-2f));
 
-        _reaction.DoEntityReaction(args.Target.Value, solution, ReactionMethod.Ingestion);
+        // Floof: favorite drinks
+        if (TryComp(args.Target, out FavoriteDrinkComponent? favoriteDrink))
+        {
+            if (!solution.TryGetReagent(new(favoriteDrink.ReagentId, null), out var quantity))
+                return;
+
+            // minimum threshold for mood effect (same as ice cream)
+            // notable difference here is that ice cream's effect occurs when metabolized
+            // whereas this occurs when ingested, so 5 is also the maximum amount you can drink at once
+            if (quantity.Quantity < 5)
+                return;
+
+            var target = args.Target.Value;
+            // raise after a timeout so the flytext doesn't overlap
+            Timer.Spawn(1000, () => RaiseLocalEvent(target, new MoodEffectEvent("FavoriteDrink")));
```

Comment:
It would be better to put all of this into a separate system and just call a method/raise an even here, but this works for now